sql/src/main/java/io/crate/analyze/AnalyzedUpdateStatement.java

```diff
@@ -53,6 +86,16 @@ public Symbol query() {
         return query;
     }
 
+    @Nullable
```

Comment:
I think nullable and empty list is semantically different. In PostgreSQL you can do `select;` and you get an empty resultset without columns. We don't support that (yet), but I think we should keep the possibility for that semantic difference. So here `null` is not `no columns` but `no result set`.

Reply:
ah i see. but would it be the same for returning clause? we cannot have insert or update statements without returning items if the returning clause is defined, so i am just not just sure whether there is a need to distinguish between a null and empty list.

Reply:
You're right, for `RETURNING` it is not allowed to specify no columns, so we could use an empty list to indicate the absence of RETURNING.

Reply:
Thanks for the feedback, in general I like the approach using an empty List would work nicely for `returnValues`. My main concern is that the rest of the code base defines not present usually as `null`. It would also not work for the output Fields, since then handling and then semantic is different here. What do you think ?

Reply:
> Another way could be to use Optional<List<Symbol>>

-1 for `Optional<List<Symbol>>`

I'd go for an empty list as @kovrus suggested if there is no semantic difference between `null` and empty list. We do have other places in the code where that is already the case. So I think

> My main concern is that the rest of the code base defines not present usually as null

shouldn't be a major concern.

Reply:
Could you please explain the semantic difference? Is there any issues in adjusting the rest of code to handle empty returning items? i left similar comments regarding null arrays and lists in other places as well.
`Optional<List<Symbol>>` - better to stay with nullable

Reply:
Awesome, let's do it. Thanks for this discussion.